pkg/reconciler/v1alpha1/clusteringress/clusteringress_test.go

```diff
@@ -657,3 +670,91 @@ func TestGlobalResyncOnUpdateGatewayConfigMap(t *testing.T) {
 		t.Error(err)
 	}
 }
+
+func TestGlobalResyncOnUpdateNetwork(t *testing.T) {
+	defer ClearAllLoggers()
+	_, sharedClient, servingClient, controller, _, _, sharedInformer, servingInformer, watcher := newTestSetup(t)
+
+	stopCh := make(chan struct{})
+	grp := errgroup.Group{}
+	defer func() {
+		close(stopCh)
+		if err := grp.Wait(); err != nil {
+			t.Errorf("Wait() = %v", err)
+		}
+	}()
+
+	h := NewHooks()
+
+	// Check for Gateway created as a signal that syncHandler ran
+	h.OnUpdate(&sharedClient.Fake, "gateways", func(obj runtime.Object) HookResult {
+		updatedGateway := obj.(*v1alpha3.Gateway)
+		// The expected gateway should include the Istio TLS server.
+		expectedGateway := gateway("knative-shared-gateway", system.Namespace(), []v1alpha3.Server{ingressTLSServer})
+		if diff := cmp.Diff(updatedGateway, expectedGateway); diff != "" {
+			t.Logf("Want Gateway %v, but got %v", expectedGateway, updatedGateway)
```

Comment:
You might consider using `%+v` or `spew` here to get better output.

Reply:
Done.